CHANGELOG.md

```diff
@@ -12,6 +12,9 @@ Unreleased` header.
 # Unreleased
 
 - On Windows, fix deadlock when accessing the state during `Cursor{Enter,Leave}`.
+- On Wayland, apply correct scale to `PhysicalSize` passed in `WindowBuilder::with_inner_size` when possible.
+- On Wayland, use `configure_bounds` to constrain `with_inner_size` when compositor wants users to pick size.
```

Comment:
I think that there should still be a way to ignore the `configure_bounds` hint. I.e. if I'm making a game and want to spawn it at an exact resolution for testing, even if larger than the working area, I should be able to do that.

Reply:
The constraint is the same as in GTK btw. Games do create in fullscreen though, so all of that is not applied to them, since fullscreen has size in configure.

Reply:
> The constraint is the same as in GTK btw

I argued with GTK devs about this too, I think it's wrong there also.

> Games do create in fullscreen though, so all of that is not applied to them, since fullscreen has size in configure.

Not necessarily. Windowed mode in games is very much a thing, there's even "borderless window" mode in many games.

Reply:
I still don't know a single game which can open large than working area. And anyway, the game can resize itself to the size it wants it to be.